Source/CarthageKit/Xcode.swift

```diff
@@ -1080,6 +1082,35 @@ public func architecturesInPackage(_ packageURL: URL) -> SignalProducer<String,
 		}
 }
 
+/// Strips debug symbols from the given framework
+public func stripDebugSymbols(_ frameworkURL: URL) -> SignalProducer<(), CarthageError> {
+	return SignalProducer<URL, CarthageError> { () -> Result<URL, CarthageError> in binaryURL(frameworkURL) }
+		.flatMap(.merge) { binaryURL -> SignalProducer<TaskEvent<Data>, CarthageError> in
+			let stripTask = Task("/usr/bin/xcrun", arguments: [ "strip", "-S","-o", binaryURL.path, binaryURL.path])
+			return stripTask.launch()
+				.mapError(CarthageError.taskError)
+		}
+		.then(SignalProducer<(), CarthageError>.empty)
+}
+
+/// Returns whether a given framework contains debug symbols
+/// There are many suggestions on how to do this but no one single
+/// accepted way. This seems to work best:
+/// https://lists.apple.com/archives/unix-porting/2006/Feb/msg00021.html
+public func packageContainsDebugSymbols(_ packageURL: URL) -> SignalProducer<Bool, CarthageError> {
```

Comment:
It looks like this is only used in the tests. Let's move it there.

Reply:
@mdiep Agree and my first attempt had it there - the one problem was I need to user `binaryURL()` which is private to Xcode.swift. 

I could make `binaryURL()` public, or I could just copy `binaryURL()` into the test. Thoughts?

Reply:
I don't see an issue with making it `public` or  `internal`